sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Internals/Statsbeat/AzureMonitorStatsbeat.cs

```diff
@@ -189,6 +189,15 @@ private void SetResourceProviderDetails(IPlatform platform)
                 return;
             }
 
+            var kubernetesSrvHost = platform.GetEnvironmentVariable(EnvironmentVariableConstants.KUBERNETES_SERVICE_HOST);
+            if (kubernetesSrvHost != null)
+            {
+                _resourceProvider = "aks";
+                _resourceProviderId = platform.GetEnvironmentVariable(EnvironmentVariableConstants.AKS_ARM_NAMESPACE_ID) ?? "unknown";
+
+                return;
+            }
```

Comment:
var aksArmNamespaceId = platform.GetEnvironmentVariable(EnvironmentVariableConstants.AKS_ARM_NAMESPACE_ID);
if (aksArmNamespaceId != null)
{
    _resourceProvider = "aks";
    _resourceProviderId = aksArmNamespaceId

    return;
}

Reply:
@rajkumar-rangaraj i'm proposing this for now.